src/Simulations/run.jl

```diff
@@ -169,9 +176,11 @@ the largest iteration.
 """
 function checkpoint_path(pickup::Bool, checkpointer::Checkpointer)
     filepaths = glob(checkpoint_superprefix(checkpointer.prefix) * "*.jld2", checkpointer.dir)
-    filenames = basename.(filepaths)
+
+    length(filepaths) == 0 && return nothing
 
     # Parse filenames to find latest checkpointed iteration
+    filenames = basename.(filepaths)
```

Comment:
Is it a little easier to understand the code if we write

```julia
if length(filepaths) == 0 # no checkpoint files found
    return nothing
else
    return latest_checkpoint(checkpointer, filepaths)
end

function latest_checkpoint(filepaths, checkpointer)
    filenames = basename.(filepaths)
    leading = length(checkpoint_superprefix(checkpointer.prefix))
    trailing = 5 # length(".jld2")
    iterations = map(name -> parse(Int, name[leading+1:end-trailing]), filenames)
    latest_iteration, idx = findmax(iterations)
    return filepaths[idx]
end
```

?

Reply:
That looks nice, thanks for the suggestion!